plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTransforms.java

```diff
@@ -373,6 +380,14 @@ private static int bucketHash(Slice value)
         return Murmur3Hash32.hash(value);
     }
 
+    private static int bucketStringHash(Slice value)
+    {
+        // We need to use slower implementation to match Iceberg
```

Comment:
Either airlift/slice and commons-codec don't hash surrogates correctly or Guava is wrong. Commons-codec's `Murmur3#hash32_x86` also produces same results as the impl in `airlift/slice`.

FWIW the Iceberg Python library doesn't allow surrogates. I'm not sure what to do here to be **spec compliant**. 🙁 

```
775615312 // Google's hash impl used in Iceberg
-53444625 // commons-codec
-53444625 // airlift slice
```

cc: @electrum

Reply:
That's why I've used the same implementation detail as Iceberg. Spec doesn't tell anything about hashing specific requirements